spring-webflux/src/main/java/org/springframework/web/reactive/socket/adapter/JettyWebSocketSession.java

```diff
@@ -108,25 +152,61 @@ public boolean isOpen() {
 	public Mono<Void> close(CloseStatus status) {
 		Callback.Completable callback = new Callback.Completable();
 		getDelegate().close(status.getCode(), status.getReason(), callback);
-
 		return Mono.fromFuture(callback);
 	}
 
+	@Override
+	public Mono<CloseStatus> closeStatus() {
+		return this.closeStatusSink.asMono();
+	}
 
-	private final class SendProcessorCallback implements Callback {
+	@Override
+	public Flux<WebSocketMessage> receive() {
+		return this.flux;
+	}
 
-		@Override
-		public void fail(Throwable x) {
-			getSendProcessor().cancel();
-			getSendProcessor().onError(x);
-		}
+	@Override
+	public Mono<Void> send(Publisher<WebSocketMessage> messages) {
+		return Flux.from(messages)
+				.flatMap(this::sendMessage, 1)
+				.then();
+	}
 
-		@Override
-		public void succeed() {
-			getSendProcessor().setReadyToSend(true);
-			getSendProcessor().onWritePossible();
-		}
+	protected Mono<Void> sendMessage(WebSocketMessage message) {
 
-	}
+		Callback.Completable completable = new Callback.Completable();
 
+		DataBuffer dataBuffer = message.getPayload();
+		Session session = getDelegate();
+		if (WebSocketMessage.Type.TEXT.equals(message.getType())) {
+			String text = dataBuffer.toString(StandardCharsets.UTF_8);
+			session.sendText(text, completable);
+		}
+		else {
+			switch (message.getType()) {
+				case BINARY -> {
+					try (DataBuffer.ByteBufferIterator iterator = dataBuffer.readableByteBuffers()) {
+						while (iterator.hasNext()) {
+							ByteBuffer byteBuffer = iterator.next();
+							session.sendBinary(byteBuffer, completable);
+						}
```

Comment:
This does not look correct?  Can you use the completable callback multiple times? Doesn't this need to be some composition of completables?

Reply:
Yeah this is wrong, and its sending multiple binary messages from the DataBuffer of a single message.

It was like this before as well, it should be only 1 BINARY WebSocket message. I will update this to use sendPartialBinary with an iterating callback.

Reply:
done